tests/bluetooth/mesh/sbr/src/main.c

```diff
@@ -0,0 +1,409 @@
+		.direction = BRG_ADDR1_TO_ADDR2,
+		.netidx1 = 1,
+		.netidx2 = 2,
+		.addr1 = 3,
+		.addr2 = 4,
+	};
+
+	int err = brg_cfg_tbl_add(test_vec.direction, test_vec.netidx1,
+			test_vec.netidx2, test_vec.addr1, test_vec.addr2);
+	zassert_equal(err, 0);
+
+	ztest_expect_data(settings_save_one, name, "bt/mesh/brgbt");
+	ztest_expect_value(settings_save_one, val_len, sizeof(test_vec));
+	ztest_expect_data(settings_save_one, value, &test_vec);
+
+	bt_mesh_brg_tbl_pending_store();
+}
+
+
+ZTEST(bt_mesh_brg_cfg, test_tbl_add_invalid_ip)
+{
+	int err;
+	/* Create test vector array of test_brg_cfg_row iteams with invalid values.
+	 * Each vector has only one invalid field value, rest all are valid values.
+	 */
+	struct test_brg_cfg_row inv_test_vector[] = {
+	/* Direction has invalid values */
+	{.direction = BRG_PROHIBITED, .netidx1 = 0, .netidx2 = 1, .addr1 = 1, .addr2 = 2},
+	{.direction = BRG_RFU_MAX, .netidx1 = 0, .netidx2 = 1, .addr1 = 1, .addr2 = 2},
+	/* Out of range netidx values */
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 4096, .netidx2 = 1, .addr1 = 1, .addr2 = 2},
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 0, .netidx2 = 4096, .addr1 = 1, .addr2 = 2},
+	/* Same netidx values */
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 0, .netidx2 = 0, .addr1 = 1, .addr2 = 2},
+	/* Same addr values */
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 1, .addr2 = 1},
+	/* Invalid address1 value */
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 0, .addr2 = 1},
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 0x8000, .addr2 = 1},
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 0xC000, .addr2 = 1},
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 0xFFFE, .addr2 = 1},
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 0xFFFF, .addr2 = 1},
+	/* Invalid address2 values */
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 1, .addr2 = 0},
+	{.direction = BRG_ADDR1_TO_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 1, .addr2 = 0xFFFF},
+	{.direction = BRG_ADDR1_TF_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 1, .addr2 = 0x8000},
+	{.direction = BRG_ADDR1_TF_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 1, .addr2 = 0xC000},
+	{.direction = BRG_ADDR1_TF_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 1, .addr2 = 0xFFFE},
+	{.direction = BRG_ADDR1_TF_ADDR2, .netidx1 = 0, .netidx2 = 1, .addr1 = 1, .addr2 = 0xFFFF},
+	};
+
+	check_brg_cfg_tbl_reset();
+
+	for (int i = 0; i < ARRAY_SIZE(inv_test_vector); i++) {
+		err = brg_cfg_tbl_add(inv_test_vector[i].direction, inv_test_vector[i].netidx1,
+				       inv_test_vector[i].netidx2, inv_test_vector[i].addr1,
+				       inv_test_vector[i].addr2);
+		zassert_equal(err, -EINVAL, "Test vector index: %zu", i);
+	}
+}
+
+
+/* Following test is written only for the purpose of checking performance on real devices. */
+#if (0)
```

Comment:
I am a little bit worried about letting this be here. Either add the tests or remove them. They do not have much value as they are now.